scripts/validate-npm-packages.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script is used to validate the npm packages that are published to npmjs.org are in the correct format.
+# It won't catch things like malformed JS or Types but it will assert that the package has
+# the correct files and package.json properties.
+ARTIFACTS_DIR="./npm-artifacts"
+
+for file in $ARTIFACTS_DIR/*.tgz; do
+  echo "🔍 Checking NPM package: $file"
+
+  dir_name=$(basename "$file" .tgz)
+  mkdir -p "./npm-artifacts/$dir_name"
+  tar -xzf "$file" -C "./npm-artifacts/$dir_name" --strip-components=1
+
+  # Navigate inside the new extracted directory
+  cd "./npm-artifacts/$dir_name"
+
+  # Check for required files and properties
+  if [ ! -f package.json ] || [ ! -f README.md ] || [ ! -f CHANGELOG.md ] || [ ! -f LICENSE_APACHE2 ]; then
+    echo -e "❌ Failed: Missing required files in directory: $dir_name.\n"
+    exit 1
+  fi
+
+  # Skip @grafana/toolkit as its structure is so different to the other packages
+  if [[ "$dir_name" == "@grafana-toolkit"* ]]; then
+
+    if [ ! -d bin ] || [ ! -f bin/grafana-toolkit.js ]; then
+      echo -e "❌ Failed: Missing 'bin' directory or required files in directory: $dir_name.\n"
+      exit 1
+    fi
+
+    echo -e "✅ Passed: package checks for $file.\n"
+    cd ../..
+    continue
+  fi
+
+  # Assert commonjs builds
+  if [ ! -d dist ] || [ ! -f dist/index.js ] || [ ! -f dist/index.d.ts ]; then
+    echo -e "❌ Failed: Missing 'dist' directory or required files in directory: $dir_name.\n"
+    exit 1
+  fi
+
+  if [ "$(jq -r '.main' package.json)" != "dist/index.js" ] || \
```

Comment:
just a tip in case `jq` is not available in CI environment. you can use a combination of unix tools or invoke an inline python script that parses the json and output the value. e.g: `cat package.json | python3 -c "import sys, json; print(json.load(sys.stdin)['main'])"`